client/blocks/product-selector/index.jsx

```diff
@@ -143,10 +144,21 @@ export class ProductSelector extends Component {
 		} );
 	}
 
-	handleCheckoutForProduct = productObject => {
-		const { selectedSiteSlug } = this.props;
+	handleCheckoutForProduct = ( productObject, product ) => {
+		const { currentPlanSlug, selectedSiteSlug } = this.props;
+
+		const tracksEventProps = {
+			upgrading_to: productObject.product_slug,
+			current_plan: currentPlanSlug,
+		};
+
+		const productPurchase = this.getPurchaseByProduct( product );
+		if ( productPurchase && productPurchase.productSlug ) {
+			tracksEventProps.current_product = productPurchase.productSlug;
+		}
 
 		return () => {
+			this.props.recordTracksEvent( 'calypso_product_feature_upgrade_click', tracksEventProps );
```

Comment:
Please note that it is intentionally using a full object path, i.e. `this.props.recordTracksEvents`. It's here to prevent ESLint's `no-shadow` error.